cpp11test/src/test-protect.cpp

```diff
@@ -1,3 +1,16 @@
+// Avoid unused variable warnings regarding `cpp11::preserved`
+// clang-format off
+#ifdef __clang__
+# pragma clang diagnostic push
+# pragma clang diagnostic ignored "-Wunused-variable"
+#endif
+
+#ifdef __GNUC__
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wunused-variable"
+#endif
+// clang-format on
```

Comment:
Another thing that showed up on the Windows build with the "latest" version of R while compiling cpp11test was a warning about `cpp11::preserved` being an unused variable in this `test-protect.cpp` file.

I guess that makes sense, because it isn't used anywhere in this test file but is created as a `static struct` in the header file so each compilation unit gets a copy of it and is "expected" to use it. Not sure why we don't get this warning on other OSes, but I just copied over the same pragma from `protect.cpp` that is also used to ignore some unused variable warnings
https://github.com/r-lib/cpp11/blob/cb1a6884990adb6d5c241e86fcfd9293e122c5fc/inst/include/cpp11/protect.hpp#L301
https://github.com/r-lib/cpp11/blob/cb1a6884990adb6d5c241e86fcfd9293e122c5fc/cpp11test/src/protect.cpp#L37

Corresponds to this commit https://github.com/r-lib/cpp11/pull/279/commits/0fbdcb1ddb35146385689a89ade4bc061ede5f75